scripts/core/editor3/helpers/editor3CustomData.js

```diff
@@ -42,10 +42,18 @@ export function setCustomDataForEditor(editorState, key, value) {
 }
 
 function getDataFromArrayState(editorStateArr, key) {
-    const editorState = editorStateArr[0];
-    const firstBlock = editorState.blocks[0];
+    let data = [];
 
-    return firstBlock.data[key];
+    // for each editor state we collect all data as one single array
+    editorStateArr.forEach((editorState) => {
```

Comment:
Do we already save the editorState for every field that uses editor3?

Reply:
We don't, the array is always one element right now (the body editor). I assumed when we have suggestions/comments in multiple fields we'll have several items in this array.